docs/source/performance/apq.md

```diff
@@ -3,78 +3,103 @@ title: Automatic persisted queries
+Add the persisted query link anywhere in the chain before the terminating link. This example shows a basic two-link chain:
 
 ```js
-import { createPersistedQueryLink } from "apollo-link-persisted-queries";
-import { createHttpLink } from "apollo-link-http";
-import { InMemoryCache } from "apollo-cache-inmemory";
-import ApolloClient from "apollo-client";
+import { ApolloClient, InMemoryCache, HttpLink } from "@apollo/client";
+import { createPersistedQueryLink } from "@apollo/client/link/persisted-queries";
 
-const link = createPersistedQueryLink().concat(createHttpLink({ uri: "/graphql" }));
+const linkChain = createPersistedQueryLink().concat(new HttpLink({ uri: "http://localhost:4000/graphql" }));
 
 const client = new ApolloClient({
   cache: new InMemoryCache(),
   link: link,
 });
 ```
 
-> Note: Users of `apollo-boost` should [migrate to `apollo-client`](https://www.apollographql.com/docs/react/advanced/boost-migration/) in order to use the `apollo-link-persisted-queries` package.
+## Command-line testing
 
-## Verify
+You can test out APQ directly from the command line. This section also helps illustrate the shape of APQ requests, so you can use it to add APQ support to GraphQL clients besides Apollo Client.
 
-Apollo Server's persisted queries configuration can be tested from the command-line. The following examples assume Apollo Server is running at `localhost:4000/`.
-This example persists a dummy query of `{__typename}`, using its sha256 hash: `ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38`.
+> This section assumes your server is running locally at `http://localhost:4000/graphql`.
 
+Almost every GraphQL server supports the following query (which requests the `__typename` field from the `Query` type):
 
-1. Request a persisted query:
+```graphql
+{__typename}
+```
 
-   ```bash
-   curl -g 'http://localhost:4000/?extensions={"persistedQuery":{"version":1,"sha256Hash":"ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38"}}'
-   ```
+The SHA-256 hash of this query string is the following:
 
-   Expect a response of: `{"errors": [{"message": "PersistedQueryNotFound", "extensions": {...}}]}`.
+```none
+ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38
+```
 
-2. Store the query to the cache:
+1. Attempt to execute this query on your running server by providing its hash in a `curl` command, like so:
 
-   ```bash
-   curl -g 'http://localhost:4000/?query={__typename}&extensions={"persistedQuery":{"version":1,"sha256Hash":"ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38"}}'
-   ```
+    ```shell
+    curl -g 'http://localhost:4000/graphql?extensions={"persistedQuery":{"version":1,"sha256Hash":"ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38"}}'
```

Comment:
I wonder if we really should be showing off GET here (with the super non-standard extensions query param) vs POST with body where you do

```
curl -H 'content-type: application/json' -XPOST -d '{"query":"{__typename}"}' http://localhost:4000/graphql
```

and 
```
curl -H 'content-type: application/json' -XPOST -d '{"extensions":{"persistedQuery":{"version":1,"sha256Hash":"ecf4edb46db40b5132295c0291d62fb65d6759a9eedfa4d5d612dd5ec54a6b38"}},"query":"{__typename}"}' http://localhost:400/graphql
```
etc

Means you don't have to use the weird `-g` flag at least... I dunno.

Reply:
Hm, I see what you mean about the `-g` option. I think for now I'll leave this as `GET`s, just to mirror the effect of the persisted query link. Will definitely revisit if we get any negative feedback.